classes/class-wc-connect-compatibility.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * A class for working around the quircks and different versions of WordPress/WooCommerce
+ */
+
+// No direct access please
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'WC_Connect_Compatibility' ) ) {
+
+	class WC_Connect_Compatibility {
+		private static $singleton;
+
+		public static function instance() {
+			if( self::$singleton === null ) {
+				self::$singleton = new WC_Connect_Compatibility();
+			}
+
+			return self::$singleton;
+		}
+
+		public function __construct() {
+		}
```

Comment:
The singleton pattern can be useful if an instance is required, but if it isn't, a class with static methods works just as well.

`WC_Connect_Compatibility::get_item_product( $order, $item );` vs. `WC_Connect_Compatibility::instance()->get_item_product( $order, $item );`.

What do you think?